src/python/pants/backend/python/targets/python_distribution.py

```diff
@@ -38,31 +36,23 @@ def __init__(self,
     :type payload: :class:`pants.base.payload.Payload`
     :param sources: Files to "include". Paths are relative to the
       BUILD file's directory.
-    :type sources: ``Fileset`` or list of strings. Must include setup.py.
-    :param compatibility: either a string or list of strings that represents
-      interpreter compatibility for this target, using the Requirement-style
-      format, e.g. ``'CPython>=3', or just ['>=2.7','<3']`` for requirements
-      agnostic to interpreter class.
+    :type sources: :class:`twitter.common.dirutil.Fileset` or list of strings. Must include
+                   setup.py.
+    :param list setup_requires: A list of python requirements to provide during the invocation of
+                                setup.py.
     """
+    if not 'setup.py' in sources:
```

Comment:
Why strip these out? Are you planning on adding setup.py checking back?

Reply:
This should be the exact same as the previous target, just without unnecessarily repeated code, which I think should be a goal. It doesn't seem like too gross of a refactoring because it's limited to a single file. We call the superclass constructor (`PythonTarget`), which already does all of the checking that was removed from this file.

Reply:
I think I remember vaguely that we don't subclass `PythonTarget` due to some method in `pex_build_util.py` using `PythonTarget` to mean something that we don't want to include. I don't know if that is still the case with the changes to that file. Will look into and leave a comment if this is ok or revert the change / do something else if not.

Reply:
Agreed on the redundancy, but I think that PythonTarget will not check for setup.py presence as of right now, so it would be good to add that back in to this one.

Reply:
I don't understand -- we are checking if `setup.py` is in sources. I can move it back to after the super constructor call  -- it's not clear which ordering would be desired here.

Reply:
Hmm, weird, I must have read this wrong. This looks good to me regarding setup.py checking.